lib/gringotts/gateways/mercadopago.ex

```diff
@@ -0,0 +1,262 @@
+defmodule Gringotts.Gateways.Mercadopago do
+  @moduledoc """
+  [mercadopago][home] gateway implementation.
+
+  For reference see [mercadopago documentation][docs].
+
+  The following features of mercadopago are implemented:
+
+  | Action                       | Method        | 
+  | ------                       | ------        | 
+  | Pre-authorize                | `authorize/3` | 
+  | Capture                      | `capture/3`   |
+  | Purchase                     | `purchase/3`  |
+  | Reversal                     | `void/2`      |
+  | Refund                       | `refund/3`    |
+
+  [home]: https://www.mercadopago.com/
+  [docs]: https://www.mercadopago.com.ar/developers/en/api-docs/
+
+  ## The `opts` argument
+
+  Most `Gringotts` API calls accept an optional `keyword` list `opts` to supply
+  optional arguments for transactions with the mercadopago
+  gateway. The following keys are supported:
+
+  | Key                      | Remark                                                                                  |
+  | ----                     | ---                                                                                     |
+  | `email`                  | Email of the customer. Type - string .                                                  |
+  | `order_id`               | Order id issued by the merchant. Type- integer.                                         |
+  | `payment_method_id`      | Payment network operators, eg. `visa`, `mastercard`. Type- string.                      |
```

Comment:
1. `payment_method_id` is no longer required in `opts`
2. `customer_id` need not be `nil` in case of a new customer, it can simply be skipped. Please re-word the sentence.
3. `order_type` and `installments` are undocumented.